pygeos/tests/test_predicates.py

```diff
@@ -112,6 +118,19 @@ def test_equals_exact_tolerance():
     assert pygeos.equals_exact(p1, p2).item() is False
 
 
+@pytest.mark.skipif(pygeos.geos_version < (3, 10, 0), reason="GEOS < 3.10")
+def test_distance_within():
+    # specifying tolerance
+    p1 = pygeos.points(50, 4)
+    p2 = pygeos.points(50.1, 4.1)
+    actual = pygeos.distance_within([p1, p2, None], p1, distance=0.05)
+    np.testing.assert_allclose(actual, [True, False, False])
+    assert actual.dtype == np.bool_
+    actual = pygeos.distance_within([p1, p2, None], p1, distance=0.2)
+    np.testing.assert_allclose(actual, [True, True, False])
+    assert actual.dtype == np.bool_
```

Comment:
Maybe add a test with a array for `distance`? (I see that this is possible from the docstring examples, and I suppose this might not be covered by the default predicate tests?)